quick_start/kusto_sample_app.py

```diff
@@ -20,394 +16,458 @@
+                if not cls.execute_control_command(kusto_client, database_name, mapping_command):
+                    print(f"Failed to create a '{ingestion_mapping_kind}' mapping reference named '{mapping_name}'. Skipping this ingestion.")
+                    return False
+        elif cls.data_format_to_is_mapping_required(data_format) and not mapping_name:
+            print(f"The data format '{data_format.name}' requires a mapping and the configuration indicates an existing mapping should be used, but none was provided. Skipping this ingestion.")
+            return False
 
+        return True
 
-def ingest_data_from_file(client: QueuedIngestClient, db: str, table: str, file_path: str, file_format: DataFormat, mapping_ref: str = None):
-    ingestion_props = IngestionProperties(
-        database=f"{db}",
-        table=f"{table}",
-        ingestion_mapping_reference=mapping_ref,
-        # Learn More: For additional information about supported data formats, see
-        #  https://docs.microsoft.com/azure/data-explorer/ingestion-supported-formats
-        data_format=file_format,
-        # Todo - Config: Setting the ingestion batching policy takes up to 5 minutes to have an effect.
-        #  For the sake of the sample we set Flush-Immediately, but in practice it should not be commonly used.
-        #  Comment the below line after running the sample for the first few times!
-        flush_immediately=True,
-    )
+    @classmethod
+    def ingest(cls, file: dict, data_format: DataFormat, ingest_client: QueuedIngestClient, database_name: str, table_name: str, mapping_name: str) -> None:
+        source_type = str(file["sourceType"]).lower()
+        uri = file["dataSourceUri"]
+        cls.wait_for_user_to_proceed(f"Ingest '{uri}' from '{source_type}'")
+        # Tip: When ingesting json files, if each line represents a single-line json, use MULTIJSON format even if the file only contains one line.
+        # If the json contains whitespace formatting, use SINGLEJSON. In this case, only one data row json object is allowed per file.
+        data_format = DataFormat.MULTIJSON if data_format == data_format.JSON else data_format
+
+        # Tip: Kusto's Python SDK can ingest data from files, blobs, open streams and pandas dataframes.
+        #  See the SDK's samples and the E2E tests in azure.kusto.ingest for additional references.
+        if source_type == "localfilesource":
+            cls.ingest_from_file(ingest_client, database_name, table_name, uri, data_format, mapping_name)
+        elif source_type == "blobsource":
+            cls.ingest_from_blob(ingest_client, database_name, table_name, uri, data_format, mapping_name)
+        else:
+            print(f"Unknown source '{source_type}' for file '{uri}'")
+
+    @classmethod
+    def ingest_from_file(cls, ingest_client: QueuedIngestClient, database_name: str, table_name: str, file_path: str,

```

Comment:
Prefer a parent of `QueuedIngestClient`, which doesn't exist. Asaf is adding one, so we should change this once it merges.